kiosk/src/Models/Kiosk.ts

```diff
@@ -137,6 +137,9 @@ export class Kiosk {
         const addedGames = this.getAllAddedGames();
         const addedGamesObjs: GameData[] = Object.values(addedGames);
         for (const game of addedGamesObjs) {
+            if (!game?.userAdded) {
+                game.userAdded = true;
+            }
```

Comment:
```suggestion
            game.userAdded = true;
```

should be fine to just mark all the games? fwiw this one caught my eye because the ?. on the previous line wouldn't actually protect us (as it would still get a null pointer exception on line 141 if `game == null`, and similar on the one a few lines down (that won't get an npe, but would probably get some weird state if `game == null` because `this.games` would end up with a nullish entry in it.

Reply:
Should we just have a null check at the start of the inside of this for loop? It would certainly give some peace of mind just to make sure that if we have a nullish game, it won't be added to the list. I'm not sure if that's possible and it thus might be adding a branch for no reason, but it might just be a protection that we want to have.